gallery/gallery/tool/grind.dart

```diff
@@ -68,6 +99,26 @@ Future<void> _runProcess(String executable, List<String> arguments) async {
   stderr.write(result.stderr);
 }
 
+// Function to make sure we capture all of the stdout.
+// Reference: https://github.com/dart-lang/sdk/issues/31666
+Future<String> _startProcess(String executable,
+    {List<String> arguments = const [], String input}) async {
+  final output = <int>[];
+  final completer = Completer<int>();
+  final process = await Process.start(executable, arguments);
+  process.stdin.writeln(input);
+  process.stdout.listen((event) {
+    output.addAll(event);
+  }, onDone: () async => completer.complete(await process.exitCode));
+  await process.stdin.close();
+
+  final exitCode = await completer.future;
+  if (exitCode != 0) {
```

Comment:
what happens if you were to call exit(0) ?

Reply:
Then we would have exited the program, even though the process was successfully run. I.e. non-intended behavior. I'll add back the line `stderr.write('Running "$executable ${args.join(' ')}" failed with $exitCode.\n');` so that it's more clear for why we have this.